src/components/nmrfile-browser/NMRFileBrowser.tsx

```diff
@@ -0,0 +1,507 @@
+  return (
+    <div
+      style={{
+        padding: '5px 0 0 0',
+        height: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+      }}
+      ref={TableRef}
+    >
+      <div
+        tabIndex={0}
+        css={css({
+          padding: '0 5px',
+          marginTop: '5px',
+          backgroundColor: 'white',
+          top: '5px',
+          display: 'flex',
+          alignItems: 'center',
+          gap: 6,
+          width: '100%',
+        })}
+      >
+        <InputGroup
+          css={css({
+            flexGrow: 1,
+          })}
+          placeholder="Search for a parameter"
+          onChange={({ target }) => {
+            if (target.value !== undefined) {
+              processQuery(target.value)
+                .then((entries) => {
+                  setEntries(entries);
+                })
+                .catch(() => {});
+            }
+          }}
+          leftIcon="search"
+          type="search"
+          fill
+        />
+        [{entries.length}/{total}]<div id="selection" />
+      </div>
+      <div
+        style={{
+          position: 'relative',
+          marginTop: '5px',
+          display: 'flex',
+          flexDirection: 'column',
+          overflowY: 'auto',
+          flex: 1,
+        }}
+      >
+        {entries.slice(0, displayFirst).map((entry: NMREntry) => (
+          <Collapsible.Root
+            key={entry.groupName}
+            className="CollapsibleRoot"
+            open={opened === entry.groupName}
+            onOpenChange={(open) => {
+              if (open) {
+                setOpened(entry.groupName);
+              } else {
+                setOpened('');
+              }
+            }}
+          >
+            <Collapsible.Trigger asChild css={style.button}>
+              <div
+                style={{
+                  display: 'flex',
+                  justifyContent: 'space-between',
+                  width: '100%',
+                  gap: '10px',
+                }}
+              >
+                <Icon icon="chevron-right" css={style.chevron} />
+                <div
+                  style={{
+                    display: 'flex',
+                    justifyContent: 'space-between',
+                    width: '100%',
+                    gap: '10px',
+                  }}
+                >
+                  <span
+                    style={{
+                      overflow: 'hidden',
+                      alignSelf: 'center',
+                      textOverflow: 'ellipsis',
+                      whiteSpace: opened === entry.groupName ? '' : 'nowrap',
+                      wordWrap: 'break-word',
+                      width: '1em',
+                      flex: 1,
+                    }}
+                  >
+                    {entry.groupName}
+                  </span>
+                  {TableRef.current?.offsetWidth === undefined ||
+                    (TableRef.current?.offsetWidth > 300 && (
```

Comment:
This is not ok, because the component is not guaranteed to be rerendered when its size changes. Please use `use-resize-observer` to watch size changes.